tests/jobs.py

```diff
@@ -851,3 +851,120 @@ def test_dag_with_system_exit(self):
         session = settings.Session()
         self.assertEqual(
             len(session.query(TI).filter(TI.dag_id == dag_id).all()), 1)
+
+    def test_dag_backfill_option(self):
+        """
+        Test to check that a DAG with no backfill only schedules beginning now, not back to the start date
+        """
+
+        now = datetime.datetime.now()
+        three_hours_ago_to_the_hour = (now - datetime.timedelta(hours=6)).replace(minute=0, second=0, microsecond=0)
+        now_minus_three_minutes = now - datetime.timedelta(minutes=3)
+        now_minus_two_hours_and_three_minutes = now_minus_three_minutes - datetime.timedelta(hours=2)
+
+        START_DATE = three_hours_ago_to_the_hour
+        DAG_NAME1 = 'no_backfill_test1'
+        DAG_NAME2 = 'no_backfill_test2'
+        DAG_NAME3 = 'no_backfill_test3'
+
+        default_args = {
+            'owner': 'airflow',
+            'depends_on_past': False,
+            'start_date': START_DATE
+
+        }
+        dag1 = DAG(DAG_NAME1,
+                  schedule_interval='* * * * *',
+                  max_active_runs=1,
+                  default_args=default_args
+                  )
+
+        default_backfill = configuration.getboolean('scheduler', 'backfill_by_default')
+        # Test configs have backfill by default ON
+        if default_backfill == 'true':
+            default_backfill = True
+        elif default_backfill == 'false':
+            default_backfill = False
+
+        self.assertEqual(default_backfill, True)
+
+        # Correct default?
+        self.assertEqual(dag1.backfill, True)
+
+        # Change the default to False and validate
+        configuration.set('scheduler', 'backfill_by_default', 'false')
```

Comment:
The whole point of this is to test the config, one way, then the other. The tests should reset the config each time, so it isn't an issue.

Not sure how we could test two config values without changing the config though. Open to ideas...